swe/morphological_options.py

```diff
@@ -62,17 +70,17 @@ def set_up_suspended(self, mesh, tracer=None):
         if self.t_old.dat.data[:] == 0.0:
             self.set_bathymetry(P1)
         else:
-            self.bathymetry = project(self.bathymetry, P1)
+            self.bathymetry = interpolate(self.bathymetry, P1)
 
-        self.depth = project(self.elev_cg + self.bathymetry, P1)
+        self.depth = interpolate(self.elev_cg + self.bathymetry, P1)
 
-        self.unorm = project(self.horizontal_velocity**2 + self.vertical_velocity**2, P1DG)
+        self.unorm = interpolate(self.horizontal_velocity**2 + self.vertical_velocity**2, P1DG)
```

Comment:
In these cases, which space are you interpolating from? How come you decided not to project?

Reply:
I realised that horizontal_velocity and vertical_velocity were both in self.P1DG too. Therefore to save computational times and reduce projection error I changed from a project to an interpolate